### ISSUE TITLE
Add "nosleep" linter

### ISSUE THREAD

**Issue by powerman** [golangci/golangci-lint#3958]
### Your feature request related to a problem? Please describe.

Using time.Sleep in tests is a mistake in most of cases.

### Describe the solution you'd like.

Linter should detect time.Sleep in tests.

### Describe alternatives you've considered.

TBH as golangci-link contains nearly hundred of linters it's hard to say is some of existing ones can be configured to do this.

### Additional context.

https://xeiaso.net/blog/nosleep

**Comment by dshemin** [comment]
Hi!

You can use [forbidigo](https://golangci-lint.run/usage/linters/#forbidigo) linter for this case.
Just put this lines to config:

```yaml
linters:
  # ...
  enable:
    # ...
    - forbidigo
    # ...
# ...
linters-settings:
  # ...
  forbidigo:
      forbid:
          - p: time.Sleep
            msg: "Avoid time.Sleep in you code please"
  #...
```

**Comment by powerman** [comment]
Isn't this will apply both to tests and code, instead of just tests?

**Comment by bombsimon** [comment]
You can use the [issues section](https://golangci-lint.run/usage/configuration/#issues-configuration) to configure files to include/exclude:

```yaml
issues:
    - path-except: _test\.go
      linters:
        - forbidigo
```

**Comment by powerman** [comment]
Yes, but this will exclude all rules of the linter, not just this one.
Probably it's possible to except using `text:` instead of `linters:`, but it feels like a workaround: error message text to match may change in future versions, linter will still spend time applying this rule to non-tests, etc.
